src/DataGrid.tsx

```diff
@@ -624,9 +625,11 @@ function DataGrid<R, SR, K extends Key>(
 
   function handleScroll(event: React.UIEvent<HTMLDivElement>) {
     const { scrollTop, scrollLeft } = event.currentTarget;
-    setScrollTop(scrollTop);
-    // scrollLeft is nagative when direction is rtl
-    setScrollLeft(abs(scrollLeft));
+    flushSync(() => {
```

Comment:
Does the prod mode also have a delay?

Reply:
Yes, you can test on https://adazzle.github.io/react-data-grid/#/common-features